File: code/noise.py
import cv2 as cv
import numpy as np
from skimage import util
from PIL import Image
import matplotlib.pyplot as plt

image=cv.imread('/disk/course/dip/term/sourceimage/lena.bmp')
noise_gaussian_image=util.random_noise(image,mode='gaussian')
noise_saltpepper_image=util.random_noise(image,mode='s&p')



cv.imshow('image',image);
cv.imshow('gauss noise image',noise_gaussian_image)
cv.imshow('salt pepper noise image',noise_saltpepper_image)

noise_gaussian_image*=256
cv.imwrite("guassian.bmp",noise_gaussian_image);

cv.waitKey(0)

File: code/noise.m

close all;
clear all;
clc;

raw_image=imread('/disk/repository/course/dip/term/sourceimage/jet.bmp');
guass_image=imnoise(raw_image,'gaussian',0,0.03);
saltpepper_image=imnoise(raw_image,'salt & pepper',0.3);
median_filter_guass_image=median_filter(guass_image,3);
median_filter_saltpepper_image=median_filter(saltpepper_image,3);

subplot(1,5,1);
imshow(raw_image);
title('image');

subplot(1,5,2);
imshow(guass_image);
title('guassian noise');

subplot(1,5,3);
imshow(saltpepper_image);
title('salt&pepper noise');

subplot(1,5,4);
imshow(median_filter_guass_image);
title('median filter for guassian');

subplot(1,5,5);
imshow(median_filter_saltpepper_image);
title('median filter for salt&pepper');